src/consent-manager-builder/index.tsx

```diff
@@ -316,8 +320,11 @@ export default class ConsentManagerBuilder extends Component<Props, State> {
 
       const newDestinations = getNewDestinations(destinations, destinationPreferences)
 
-      // If preferences haven't changed, don't reload the page as it's a disruptive experience for end-users
-      if (prevState.havePreferencesChanged || newDestinations.length > 0) {
+      if (
+        prevState.havePreferencesChanged ||
+        newDestinations.length > 0 ||
+        typeof newPreferences === 'boolean'
```

Comment:
We have to update the `handleSaveConsent` function signature to allow for `boolean` ... so something like this: 
```
handleSaveConsent = (newPreferences: CategoryPreferences | undefined | boolean, shouldReload: boolean)
```

Reply:
Agree with you.